File: frontend/src/api/digest.ts
import api from './client';

export interface ArticleSummary {
  id: string;
  title: string;
  source_name: string | null;
  source_url: string;
  tl_dr: string | null;
  key_takeaways: string[] | null;
  bias_rating: string | null;
  estimated_read_time_minutes: number | null;
  published_at: string | null;
  categories: string[];
  relevance_score: number;
  is_read: boolean;
  impact_radar: string[];
}

export interface DigestResponse {
  id: string;
  digest_type: string;
  title: string | null;
  content: string;
  articles: ArticleSummary[];
  generated_at: string;
  delivered_at: string | null;
}

export async function getDigest(type: string = 'daily', topic?: string): Promise<DigestResponse> {
  const params: Record<string, string> = { type };
  if (topic) params.topic = topic;
  const { data } = await api.get('/digest', { params });
  return data;
}

export async function getArticles(params?: {
  skip?: number;
  limit?: number;
  max_minutes?: number;
}): Promise<ArticleSummary[]> {
  const { data } = await api.get('/digest/articles', { params });
  return data;
}

export async function getArticleTLDR(articleId: string): Promise<any> {
  const { data } = await api.get(`/digest/articles/${articleId}/tldr`);
  return data;
}

File: backend/app/schemas/digest.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import datetime, timezone


class ArticleSummary(BaseModel):
    id: str
    title: str
    source_name: Optional[str] = None
    source_url: str
    tl_dr: Optional[str] = None
    key_takeaways: Optional[List[str]] = None
    bias_rating: Optional[str] = None
    estimated_read_time_minutes: Optional[int] = None
    published_at: Optional[datetime] = None
    categories: List[str]
    relevance_score: float = 0.0
    is_read: bool = False
    impact_radar: List[str] = []


class DigestResponse(BaseModel):
    id: str
    digest_type: str
    title: Optional[str] = None
    content: str
    articles: List[ArticleSummary]
    generated_at: datetime
    delivered_at: Optional[datetime] = None


class DigestRequest(BaseModel):
    type: str = "daily"  # "daily" | "on_demand"
    topic: Optional[str] = None


class TLDRResponse(BaseModel):
    article_id: str
    title: str
    tl_dr: str
    key_takeaways: List[str]


class ReadingTimeFilter(BaseModel):
    max_minutes: int = 5